### PR TITLE
Secure Connector Service generate token mutations

### PR DESCRIPTION
**Pull request by franpog859**
**Description**

Changes proposed in this pull request:

- Divide Connector API, and separate generate token mutation from the rest of the API
- Expose generate Token mutations within a cluster by K8S service
- Configure proper rules limiting access to Generate Token mutation so that only App Director can only access it applied

**Related issue(s)**

See the issue #302 which is a follow up for #279 

**Remember** to change `tests/connector-tests/Gopkg.toml` file after the merge of this PR. It should not point to my fork before closing [the issue](https://github.com/kyma-incubator/compass/issues/302)

### CODE REVIEW

File: components/connector/pkg/graphql/internalschema/config.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+exec:
+  filename: schema_gen.go
+  package: internalschema
+
+# Let gqlgen know where to the generated models (if any)
+model:
+  filename: models_gen.go
+  package: internalschema
+
+# Optional turns on binding to field names by tag provided
```
Comment (majakurcius) [inline components/connector/pkg/graphql/internalschema/config.yaml]:
```suggestion
# Optional enables binding to field names by tag provided
```

Comment (franpog859) [reply components/connector/pkg/graphql/internalschema/config.yaml]: Done 😆

File: components/connector/cmd/main.go [hunk 2]
```diff
@@ -166,11 +176,34 @@ func prepareGraphQLServer(cfg config, tokenResolver api.TokenResolver, certResol
...
+
+	internalExecutableSchema := internalschema.NewExecutableSchema(gqlInternalCfg)
+
+	internalRouter := mux.NewRouter()
+	internalRouter.HandleFunc("/", handler.Playground("Dataloader", cfg.PlaygroundAPIEndpoint))
+	internalRouter.HandleFunc(cfg.APIEndpoint, handler.GraphQL(internalExecutableSchema))
+
+	authContextMiddleware := authentication.NewAuthenticationContextMiddleware()
+
+	internalRouter.Use(authContextMiddleware.PropagateAuthentication)
```
[the remark this answers is not in the document]
Comment (franpog859) [reply components/connector/cmd/main.go]: Sure, we definitely don't need it here. Removed!

File: chart/compass/charts/director/templates/deployment.yaml [hunk 3]
```diff
@@ -70,7 +70,7 @@ spec:
             - name: APP_ONE_TIME_TOKEN_URL
               value: "http://compass-connector-internal.{{ .Release.Namespace }}.svc.cluster.local:{{ .Values.global.connector.graphql.internal.port }}/graphql"
             - name: APP_CONNECTOR_URL
-              value: "http://compass-connector.{{ .Release.Namespace }}.svc.cluster.local:{{ .Values.global.connector.graphql.external.port }}/graphql"
+              value: "https://{{ .Values.global.gateway.mtls.host }}.{{ .Values.global.ingress.domainName }}/connector/graphql"
```
Comment (Szymongib) [inline chart/compass/charts/director/templates/deployment.yaml]: Please use `tls` host instead of `mtls`

**Pull request merged**